Migrate Glide to Coil for image fetching

DevJethava opened the pull request:
## 📌 Description
- Migrate Glide to Coil for image fetching

## ✅ Changes
- Migrate Glide to Coil for image fetching

## 📸 Screenshots / Demo (if applicable)
| Before | After |
|--------|-------|
<img height="300" width="200" alt="Screenshot_1758563847" src="https://github.com/user-attachments/assets/dce2be03-3dee-47b3-af63-b50f02d5981a" />
<img height="300" width="200" alt="Screenshot_1758563834" src="https://github.com/user-attachments/assets/c722e696-5134-4b76-ba56-cfeb81b45056" />

## 🔗 Issues
https://github.com/hieuwu/android-groceries-store/issues/277

DevJethava committed:
- config studio

DevJethava committed:
- remove glide lib
- add coil lib
- migrate glide to coil

DevJethava committed:
- code optimization

hieuwu reviewed (changes requested):
Hi @DevJethava , please revert your credentials change in local.properties. Better add this file into .gitignore file.

Please review feedbacks from Gemini, if it’s false alarm, explain it so that I can check and resolve.

I am checking egilibility for Hacktoberfesr this year, so if feasible, I’ll mark this as Hacktoberfest-accepted

DevJethava committed:
- revert supabase keys

DevJethava committed:
- add modifier to loading state into WebImag

DevJethava commented:
Hey, @hieuwu  I’ve completed all the changes.
- Reverted the credentials in `local.properties` and added the file to `.gitignore`.
- Went through Gemini feedback - I missed out on the loading state 

Please check once and let me know if anything else is required.
Also, if it works out, it’d be great if you could mark this as **Hacktoberfest-accepted** 🙂

hieuwu reviewed (approved):
Looks good to me. Thank you for your contribution!
